src/foam/u2/detail/SectionView.js

```diff
@@ -38,37 +55,74 @@ foam.CLASS({
       class: 'Boolean',
       name: 'showTitle',
       value: true
+    },
+    {
+      class: 'FObjectProperty',
+      of: 'foam.core.ArraySlot',
+      name: 'visibilitySlots_',
+      documentation: `
+        An array of slots used to control the visibility of the child
+        SectionedDetailPropertyViews of this view. Used internally to figure out
+        which children are the first and last ones visible so we can adjust the
+        padding on those elements to make sure nested
+        SectionedDetailPropertyViews don't cause unnecessary padding.
+      `,
+      factory: function() {
+        return this.ArraySlot.create();
+      }
+    },
+    {
+      class: 'Int',
+      name: 'firstVisibleIndex_',
+      expression: function(visibilitySlots_) {
```

Comment:
Instead of these expressions, what if you do this in initE()
```
var vSlot = foam.core.ProxySlot({delegate$: this.visibilitySlots_$});
this.onDetach(this.firstVisibleIndex_$.follow(vSlot.map(ar => ar.indexOf(true)))
this.onDetach(this. lastVisibleIndex_ $.follow(vSlot.map(ar => ar.lastIndexOf(true)))
```

Reply:
What's the advantage of doing it that way over the current way? Currently it's clear from the property's definition that it's value is an expression of `visibilitySlots_` and it also avoids the overhead of the ProxySlot.

Reply:
It's currently a one-time get. You're not actually subscribing to the visibilitySlots_

Reply:
Resolved by a40bdbd.